### ISSUE TITLE
Puma-dev hangs after loading on macOS Sierra unless compiled with go 1.6.3

### ISSUE THREAD

**Issue by idyll** [puma/puma-dev#18]
I get the the point where I see `Use Ctrl-C to stop` and that's where things stop.

The page doesn't open in a Browser and it doesn't stop when I `Ctrl-C`. Instead I have to kill it.

Here's my output:

```
* Directory for apps: /Users/madsen/.puma-dev
* Domains: pdev
* DNS Server port: 9253
* HTTP Server port: 9280
! Puma dev listening
! Booted app 'tex' on socket /Users/madsen/.puma-dev/tex/tmp/puma-dev-23072.sock
tex[23075]: Puma starting in single mode...
tex[23075]: * Version 3.6.0 (ruby 2.2.5-p319), codename: Sleepy Sunday Serenity
tex[23075]: * Min threads: 0, max threads: 5
tex[23075]: * Environment: development
/Users/madsen/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/dragonfly-0.9.14/lib/dragonfly/data_storage/s3data_store.rb:27: warning: duplicated key at line 28 ignored: "sa-east-1"
tex[23075]: * Listening on unix:/Users/madsen/.puma-dev/tex/tmp/puma-dev-23072.sock
tex[23075]: Use Ctrl-C to stop
```

**Comment by evanphx** [comment]
How are you access it? Is that running via `-install`?

**Comment by idyll** [comment]
In this case no. I just compiled and ran it.

I will try with install next.

**Comment by evanphx** [comment]
What port did you access it on then? Was it `9280` because that's the port that it used when it's run in the foreground (this is due to launchd being used to control access to port 80).

**Comment by idyll** [comment]
Moved to `-install`.

Still hangs with this output:

```
tex[22226]: Puma starting in single mode...
tex[22226]: * Version 3.4.0 (ruby 2.2.3-p173), codename: Owl Bowl Brawl
tex[22226]: * Min threads: 0, max threads: 5
tex[22226]: * Environment: development
2016/07/29 16:48:26 http: proxy error: net/http: request canceled
* Directory for apps: /Users/madsen/.puma-dev
* Domains: pdev
* DNS Server port: 9253
* HTTP Server port: inherited from launchd
! Puma dev listening
* Directory for apps: /Users/madsen/.pow
* Domains: dev
* DNS Server port: 9253
* HTTP Server port: inherited from launchd
* HTTPS Server port: inherited from launchd
! Puma dev listening on http and https
2016/07/29 18:17:44 http: TLS handshake error from 127.0.0.1:61531: remote error: unknown certificate authority
! Booted app 'tex' on socket /Users/madsen/.pow/tex/tmp/puma-dev-26401.sock
bash: no job control in this shell
tput: No value for $TERM and no -T specified
tput: No value for $TERM and no -T specified
tput: No value for $TERM and no -T specified
tput: No value for $TERM and no -T specified
tput: No value for $TERM and no -T specified
tput: No value for $TERM and no -T specified
tput: No value for $TERM and no -T specified
tex[26436]: Puma starting in single mode...
tex[26436]: * Version 3.6.0 (ruby 2.2.5-p319), codename: Sleepy Sunday Serenity
tex[26436]: * Min threads: 0, max threads: 5
tex[26436]: * Environment: development
/Users/madsen/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/dragonfly-0.9.14/lib/dragonfly/data_storage/s3data_store.rb:27: warning: duplicated key at line 28 ignored: "sa-east-1"
tex[26436]: * Listening on unix:/Users/madsen/.pow/tex/tmp/puma-dev-26401.sock
tex[26436]: Use Ctrl-C to stop
```

**Comment by idyll** [comment]
I am going to grab another computer with OSX 10.11 on it and see if this works there for me.

**Comment by idyll** [comment]
So it's macOS Sierra specific. It works just fine on OSX 10.11...

How can I help debug this? I am not super familiar with godebug...

**Comment by evanphx** [comment]
I'm using the Go 1.6.2. There is a timing related bug in Go on Sierra, I know that for sure. It's possible that is causing an issue...

Do you see the request come into the app? Can you add some logging to the app to show the request coming in? That might help us find out if it's a problem with sending data back.

**Comment by idyll** [comment]
I will recompile with Go 1.6.3 locally with your https changes. This seems eerily similar to an issue that the guys from Convox were having where their connections would hang after about 15 seconds when doing deploys from Sierra.

If that doesn't work I will add extra logging.

**Comment by idyll** [comment]
Yeah Go 1.6.3 fixes it.

I didn't know how to use gb correctly and was actually building it with Go 1.6.2 like you.
Building with 1.6.3 resolves the issue.

**Comment by evanphx** [comment]
Great! I'll release v0.5 compiled with 1.6.3.

**Comment by evanphx** [comment]
Er, v0.6.

**Comment by evanphx** [comment]
Yup, here is the fix that went into 1.6.3: https://github.com/golang/go/commit/5164532ae0e9c3e87084638913029392ecc1af42
